GLAA.Web/Controllers/PublicRegisterController.cs

```diff
@@ -0,0 +1,161 @@
+
+                publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel =
+                    newSearchViewModel ?? searchViewModel;
+            }
+            else
+                searchViewModel.SearchActive = false;
+
+            //Ensure we remove any countries that are selected, from the list than are selectable.
+            if (searchViewModel.CountriesSelected != null)
+                searchViewModel.AvailableCountries = searchViewModel.AvailableCountries
+                    .Where(x => !searchViewModel.CountriesSelected.Contains(x.Value)).ToList();
+            else
+                searchViewModel.AvailableCountries = searchViewModel.AvailableCountries.ToList();
+
+            return View(publicRegisterLicenceListViewModel);
+        }
+
+        [HttpGet]
+        public IActionResult Index()
+        {
+            var licences = publicRegisterViewModelBuilder.BuildEmptySearch();
+
+            return View(licences);
+        }
+
+        [HttpGet]
+        [Route("PublicRegisterProfile/License/{id}")]
+        public IActionResult Licence(int id)
+        {
+            var licence = publicRegisterViewModelBuilder.BuildLicence(id);
+
+            return View(licence);
+        }
+
+        private PublicRegisterLicenceListViewModel AddCountry(PublicRegisterLicenceListViewModel publicRegisterLicenceListViewModel)
+        {
+            if (publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.CountriesSelected == null)
+                publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.CountriesSelected =
+                    new List<string>();
+
+            var countryAdded = publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.CountryAdded;
+
+            //add the country into the list of selected countries
+            publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.CountriesSelected.Add(
+                publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.AvailableCountries
+                    .FirstOrDefault(x => x.Value == countryAdded)?.Value);
+
+            //remove them from the list of selectable countries
+            foreach (var country in publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.CountriesSelected.ToList())
+            {
+                publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.AvailableCountries = RemoveFromList(
+                    publicRegisterLicenceListViewModel.PublicRegisterSearchViewModel.AvailableCountries,
+                    x => x.Value == country);
+            }
+
+            publicRegisterLicenceListViewModel = HandleUkSelected(publicRegisterLicenceListViewModel, countryAdded);
+
+            return publicRegisterLicenceListViewModel;
+        }
+
+        private PublicRegisterLicenceListViewModel HandleUkSelected(PublicRegisterLicenceListViewModel publicRegisterLicenceListViewModel, string countryAdded)
+        {
+            if (countryAdded == "UK")
```

Comment:
Can these be Enums?

Reply:
Going to avoid using enums for now until the country stuff is properly defined.